frontend/src/views/camp/Invitation.vue

```diff
@@ -54,31 +80,76 @@ export default {
   props: {
     campCollaboration: { type: Function, required: true }
   },
-  data: () => ({ invitationFound: undefined }),
+  data: () => ({
+    states: {
+      INIT: {},
+      INVITATION_FOUND: {},
+      INVITATION_NOT_FOUND: {},
+      REJECTED: {},
+      ERROR: {}
```

Comment:
Alle Daten in `data` werden von Vue reaktiv gemacht, also vorbereitet auf Mutationen und Observers. Konstanten gehören meiner Meinung nach stattdessen ausserhalb der Komponenten-Objektstruktur hin, auf Zeile 76.

Für diesen Fall hier habe ich in Vue aber eher öfter ein anderes Pattern gesehen als die Enum-artigen States die du verwendest. Wo immer möglich werden Probleme möglichst funktional statt imperativ gelöst:
* Status `INIT` wird errechnet als `this.campCollaboration()._meta.loading`, entweder inline oder in einer `computed`
* Status `INVITATION_FOUND` ist der Normalzustand, der angenommen wird wenn keine Spezialfälle greifen
* Für die Stati `INVITATION_NOT_FOUND` und `REJECTED` würde ich separate views mit je einer Route einrichten, und zum gegebenen Zeitpunkt dorthin redirecten (`this.$router.push(...)`)
* Status `ERROR` könnte man an einer `data`-Variablen `error` erkennen, die im Fehlerfall mit der Exception befüllt wird
* `dialogState` und `states` kann man schliesslich entfernen

Was denkst du zu dieser Idee?

Reply:
so war es vorher, war aber meiner meinung nach komplizierter, da die states dann teilweise kombiniert waren.
Wirklich funktional war es auch nicht, da es immer von state abhing, der direkt in der komponente mutable war.

Weiter gab es unschöne fehler in der konsole, wenn ich gewisse requests ausgeführt habe ohne error handler.
(this.api.get().profile(), this.campCollaboration()._meta.load)

Ich wollte die konstanten erst ausserhalb haben, aber dann kann man nicht im template darauf zugreifen.
Man könnte für jeden state eine Wrapper computed funktion haben, damit ist aber auch nicht wirklich geholfen.

Wollen wir das mal in einer session anschauen?

Reply:
Wechsel auf berechung im backend implementiert